### PR TITLE
Test CLI exit status

### PR DESCRIPTION
**Pull request by plastikos**
### What does this PR do?

Improve CLI testing and add exit status checks to various programs: salt-run, salt-syndic, salt, salt-call
### What issues does this PR fix or reference?

Inconsistent exit status
### Previous Behavior

No exit status tests
### New Behavior

Added exit status tests
### Tests written?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: tests/integration/utils/testprogram.py [hunk 1]
```diff
@@ -764,7 +765,19 @@ def shutdown(self, signum=signal.SIGTERM, timeout=10):
                         break
                     raise
                 if datetime.now() > future:
-                    raise TimeoutError('Timeout waiting for "{0}" pid'.format(pid))
```
Comment (plastikos) [inline tests/integration/utils/testprogram.py]: This change should have gone into one of the other patches or into its own patch.  Oh well.

**Comment by plastikos** [comment]
I'd love to fix the pylint failures, but I just don't see any commas on the lines noted by "PEP8 E231: missing whitespace after ','" messages.

**Comment by plastikos** [comment]
Ha!  I didn't have ``pep8`` installed on my system so I was getting the specific pylint messages when I tested locally!  Apparently the ``missing-whitespace-after-comma`` message also applies to colons in dictionaries.

Fixed.

**Comment by plastikos** [comment]
Looks like ``salt-proxy`` has a failure in providing the correct exit status - I'll add a fix for it in this P.R. so hold off just a bit before merging:

https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/3881/testReport/junit/integration.shell.proxy/ProxyTest/test_exit_status_unknown_user/

**Comment by cachedout** [comment]
@plastikos Nice find! Let me know when you're ready to go here, please.

**Comment by plastikos** [comment]
I believe this is ready.

**Comment by cachedout** [comment]
Thanks, @plastikos. Once the tests run finishes, I'll have a look. Much appreciated!

**Comment by cachedout** [comment]
This is really nice. ++.

**Comment by plastikos** [comment]
More to come along these lines in another branch . . . hopefully today.

**Pull request merged**